server/schedulers/balance_plan.go

```diff
@@ -120,23 +120,20 @@ func BalancePlanSummary(plans []plan.Plan) (map[uint64]plan.Status, bool, error)
 			return nil, false, errs.ErrDiagnosticLoadPlanError
 		}
 		step := p.GetStep()
-		if !p.status.IsNormal() {
-			normal = false
-		}
-		// we don't consider the situation for createOperator step
-		if step > shouldBalance {
-			continue
-		}
-		// We can think of shouldBalance as a filtering step for target, except that the current implementation is separate.
-		if step == shouldBalance {
+		// We can simply think of createOperator as a filtering step for target in BalancePlanSummary.
+		if step > pickTarget {
 			step = pickTarget
 		}
 		if step > maxStep {
 			storeStatusCounter = make(map[uint64]map[plan.Status]int)
 			maxStep = step
+			normal = true
 		} else if step < maxStep {
 			continue
 		}
+		if !p.status.IsNormal() {
+			normal = false
```

Comment:
Will it be changed multiple times between true and false?

Reply:
Yes. If we pick a hot region (step 1), normal is false. Then next plan is in step 2, the previous false value will be cleared up.